### PR TITLE
FEATURE: BB-531 Support table totals to be defined by AFM and ResultSpec

### PR DESCRIPTION
**Pull request by tomassatinsky**
AFM Table component generates the "totals" prop based on resultSpec

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/343/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/263/ : SUCCESS in 1m 29s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/65/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/344/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/86/ : UNSTABLE in 4m 22s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/345/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/264/ : SUCCESS in 1m 33s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/66/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/346/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/87/ : SUCCESS in 3m 07s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/350/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/88/ : FAILURE in 7m 21s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/349/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/267/ : SUCCESS in 6m 23s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/351/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/89/ : FAILURE in 5m 52s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/69/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/352/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/90/ : SUCCESS in 3m 06s

File: src/components/afm/Table.tsx [hunk 1]
```diff
@@ -21,4 +24,48 @@ function generateDefaultDimensions(afm: AFM.IAfm): AFM.IDimension[] {
...
+        resultSpectGrandTotals.forEach((resultSpectGrandTotal: AFM.ITotalItem) => {
+            const measureIndex = findIndex(afm.measures, (measure: AFM.IMeasure) => {
+                return (measure.localIdentifier === resultSpectGrandTotal.measureIdentifier);
+            });
+
+            if (measureIndex === -1) {
+                return;
+            }
+
+            const indexedTotal = find(indexedTotals, r => (r.type === resultSpectGrandTotal.type));
```
Comment (martinmosko) [inline src/components/afm/Table.tsx]: maybe more suitable variable name instead of 'r'

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/353/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/268/ : SUCCESS in 3m 09s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/70/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/354/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/91/ : SUCCESS in 2m 48s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/111/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/282/ : SUCCESS in 2m 41s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/75/ : SUCCESS in 2m 29s

**Pull request merged**